### ISSUE TITLE
Could you publish the training process?

### ISSUE THREAD

**Issue by Ernie1** [signatrix/efficientdet#10]
Hi, thanks for your excellent work! However, I run your code and it is difficult to reach mAP 0.314. I have run 2 days on 4 GPUs and the current result is only 0.209. Could you publish the loss curve or some other details? Thank you!

**Comment by uvipen** [comment]
Hi,
Thanks for your interest to our work. We run 9 days on 3 GPUs. It does take time. Unfortunately we did not record tensorboard last time, but we will do the next time when we train D1. Do you change anything in our code?

**Comment by Ernie1** [comment]
Hi, thanks for your reply! I didn't change anything, except I run on 4 GPUs.
But could you tell us how many epochs you have trained? Thanks!

**Comment by uvipen** [comment]
Hi, we trained it for ~130 epochs.

**Comment by Ernie1** [comment]
Thanks a lot!

**Comment by uvipen** [comment]
you are welcome!

**Comment by wanggrun** [comment]
Hi, thanks for the great repro!

When I trained this repro for efficientD0 with the following hyper-parameters (exactly the default hyper-parameters provided in you code), I got and IoU=0.50:0.95 of 0.21, which is far worse than the performance you reporeted in your table (.021 vs. 0.31). Could you provide the training hyper-parameters you actually used? Thank you!

python3 train.py  --image_size 512 --batch_size 8  --alpha 0.25  --gamma 1.5   --num_epochs 500  --test_interval 1 --es_min_delta 0  --es_patience 0

**Comment by uvipen** [comment]
Hi,
The parameters we set in script is the one we used. How many epochs did you train ?

**Comment by wanggrun** [comment]
I trained for 130 epochs as suggested by you.

**Comment by jGsch** [comment]
@wanggrun 

> python3 train.py --image_size 512 --batch_size 8 --alpha 0.25 **--gamma 1.5** --num_epochs 500 --test_interval 1 --es_min_delta 0 --es_patience 0

You set the `gamma` parameter to `1.5`, but it's hard codded at `2` in the loss function. Maybe that's it. 

![Screenshot from 2020-01-23 11-06-24](https://user-images.githubusercontent.com/22634019/72975164-9fb2ab80-3dd0-11ea-913a-06f628d44c93.png)

**Comment by wuhy08** [comment]
Hello @uvipen @wanggrun @jGsch Could anyone confirm if it is due to different gamma value?

Thank you!

**Comment by wuhy08** [comment]
Are we supposed to use gamma=2 or 1.5?

**Comment by uvipen** [comment]
Hi, We used gamma = 2. Sorry for unclear code (2. was used for retina and 1.5 was used for EfficientDet, but here we used 2. for EfficientDet).

**Comment by elv-xuwen** [comment]
> Hi,
> The parameters we set in script is the one we used. How many epochs did you train ?

Hi, thanks for your repo!
@Ernie1  and I have the similar issue here, I didn't change any parameters in your code but I cannot reproduce the results you publish in the table. The results from my model is only 0.215, which is far off from yours. The problem isn't the gamma value based on your discussion above because I used gamma = 2.. Do you have any ideas how it happened?
Thank you so much!

**Comment by qwasdf9** [comment]
I also tried to train with 2 Titan RTX with batch size 24 for 1.5 days, and didn't change any other parameters. Result of my model is 0.208

**Comment by lampsonSong** [comment]
Hi all, I have tried the focal loss gamma with 1.5 and 2, both DONOT work. for gamma=2, the model reaches 0.22, for gamma=1.5, the model reaches 0.207. Maybe we should try some other solutions.
